src/Tokenizer.cc

```diff
@@ -96,6 +104,22 @@ namespace onmt
       line += word;
     }
 
+    if (_spacer_annotate)
```

Comment:
Currently, this method can detokenize joiners even though `_joiner_annotate` is `false`. Would it be possible to do the same for the spacers (i.e. take action if the character is found, not if the flag is set)? Even better if this happens in the main detokenization loop above.